### PR TITLE
docs: Add API version label to search result item

### LINKED ISSUE #10311: Better handle v2/v3 split in docs search
When searching the docs all the protobuf entries show up twice with little indication that one is V2 and the other is V3 (besides a small reference in the preview text):

image

(https://www.envoyproxy.io/docs/envoy/latest/search.html?q=cluster&area=default#)

It would be nice to better indicate what the difference between these results are

### PR DESCRIPTION
**Pull request by dio**
Commit Message: This adds API version labels to search result items. This overrides the sphinx-docs' `searchtools.js`.

Additional Description: 

A sample of search result page: https://379378-65214191-gh.circle-artifacts.com/0/generated/docs/search.html?q=route&check_keywords=yes&area=default

Depends on: https://github.com/envoyproxy/envoy/pull/13008.

Risk Level: Low, cosmetics.
Testing: N/A
Docs Changes: N/A
Release Notes: N/A

Fixes #10311

Signed-off-by: Dhi Aurrahman <dio@tetrate.io>

### CODE REVIEW

**Comment by dio** [comment]
Example of search results:

**Comment by htuch** [comment]
@dio I added "Fixed #10311" to the top-level comment, since I think this is a solution for that, nice.

File: docs/root/_static/searchtools.js [hunk 1]
```diff
@@ -0,0 +1,584 @@
...
+ *
+ * Sphinx JavaScript utilities for the full-text search.
+ *
+ * :copyright: Copyright 2007-2020 by the Sphinx team, see AUTHORS.
+ * :license: BSD, see LICENSE for details.
+ *
+ */
+
+// Modified from https://raw.githubusercontent.com/sphinx-doc/sphinx/3.x/sphinx/themes/basic/static/searchtools.js
+// to have renderApiVersionLabel to render the API version for each search result item.
```
Comment (htuch) [inline docs/root/_static/searchtools.js:13]: @dio could you do a baseline PR to check in the original searchtools.js, that way we can review just the delta here?

**Comment by htuch** [comment]
FYI @snowp

File: docs/root/_static/searchtools.js (hunk 1 above)
Comment (dio) [reply docs/root/_static/searchtools.js:13]: Got it. https://github.com/envoyproxy/envoy/pull/13008.

**Comment by dio** [comment]
The latest generated docs site: https://380399-65214191-gh.circle-artifacts.com/0/generated/docs/search.html?q=cluster&check_keywords=yes&area=default

**Comment by htuch** [comment]
LGTM, but I disclaim any knowledge of CSS.

@dio looking at the rendered results at https://380399-65214191-gh.circle-artifacts.com/0/generated/docs/search.html?q=cluster&check_keywords=yes&area=default and the CSS, it looks like you are also modifying the test of the doc style. Is there a reason we need to do that in this PR?

**Comment by dio** [comment]
It is not. The only entry that is mandatory is:  `.api-version-label` I will remove the rest and pick a neutral color for api version label.

File: docs/root/_static/css/envoy.css [hunk 2]
```diff
@@ -14,3 +14,12 @@ table.docutils div.line-block {
   overflow-wrap: break-word;
   max-width: 1000px;
 }
+
+/* To style the API version label of a search result item */
+.api-version-label {
+  border-radius: 20%;
+  background-color: #c0c0c0;
```
Comment (dio) [inline docs/root/_static/css/envoy.css:21]: "silver". We can always change it later when we want to re-style the entire docs.

image

**Comment by dio** [comment]
@htuch it is updated, see the rendered docs here: https://380742-65214191-gh.circle-artifacts.com/0/generated/docs/search.html?q=cluster&check_keywords=yes&area=default

**Review by htuch — APPROVED** [review]
LGTM, thanks! I think if you do want a separate PR delving into color schemes that's totally legit, but that's the bike shed painting :)

**Pull request merged**